packages/gatsby-theme-newrelic/src/components/Collapser.js

```diff
@@ -1,16 +1,23 @@
-import React, { useState, useMemo } from 'react';
+import React, { useState, useMemo, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { css } from '@emotion/core';
 import Icon from './Icon';
 import { animated, useSpring } from 'react-spring';
 import { usePrevious, useIsomorphicLayoutEffect } from 'react-use';
 import useKeyPress from '../hooks/useKeyPress';
+import useQueryParams from '../hooks/useQueryParams';
 
 const ResizeObserver = global.ResizeObserver || class ResizeObserver {};
 
 const Collapser = ({ title, id, defaultOpen, children }) => {
+  const { queryParams } = useQueryParams();
   const [element, ref] = useState();
-  const [isOpen, setIsOpen] = useState(defaultOpen);
+  const [isOpen, setIsOpen] = useState(() => {
+    return queryParams.has('collapsers') &&
+      queryParams.get('collapsers') === 'openAll'
```

Comment:
```suggestion
      queryParams.get('collapsers') === 'open'
```

I'd suggest use using `open`, or maybe even `1` to signify "truthy" (which I've seen, for example GitHub allows "w=1" to ignore whitespace). Using a query param is usually signal that it applies to everything (hence why I'd at least drop "all" from the value)

Reply:
> @roadlittledawn this should help to fix the tests (Gatsby uses Reach Router under the hood): https://reach.tech/router/api/LocationProvider
> 
> The key is to use "memory history" as part of the router and wrap the component in the test with that `LocationProvider`. The memory history has the same API has browser history, but as it states, "its mostly used for tests". Try that out and see if you can get the tests to pass!

oh just saw this right after i pushed it up. will look and see if i missed anything but the tests succeed now. and yep, was thinking i'll add a test for query param as well.

Reply:
gotcha. i went with `open`